pkg/flow/delegate.go

```diff
@@ -0,0 +1,125 @@
+package flow
+
+import (
+	"context"
+	"net/http"
+	"path"
+
+	"github.com/gorilla/mux"
+	"github.com/grafana/agent/component"
+	"github.com/grafana/agent/pkg/cluster"
+	"github.com/grafana/agent/pkg/flow/logging"
+	"github.com/grafana/agent/web/api"
+	"github.com/prometheus/client_golang/prometheus"
+	"go.opentelemetry.io/otel/trace"
+)
+
+type delegate struct {
+	f *Flow
+	o *delegateOptions
+}
+
+type delegateOptions struct {
+	ID string
+	*moduleOptions
+}
+
+var (
+	_ component.ModuleDelegate = (*delegate)(nil)
+)
+
+// newDelegate creates a module delegate for a specific component.
+func newDelegate(id string, o *moduleOptions) *delegate {
+	return &delegate{
+		o: &delegateOptions{
+			ID:            id,
+			moduleOptions: o,
+		},
+	}
+}
+
+// LoadConfig parses River config and loads it.
+func (c *delegate) LoadConfig(config []byte, args map[string]any, onExport component.Export) error {
```

Comment:
There's a surprising behavior here: callers are expected to provide a `onExport` value every time `LaodConfig` is called, but only the first `onExport` value from the first call is used. If users try to be clever and supply different values on each call, they will be surprised to learn the value is ignored on subsequent calls. 

IMO `onExport` should be passed when the module is created, not every time when the config is loaded.

Reply:
I concur, I think are some point we should revisit `export` and how it is passed to `flow` since it has always felt a little odd. IMO at some point I think we may need to pass in the calling component as an interface and allow `module.module` to access id, export and other fields as needed but outside the scope of this change.